components/board/genericlinux/pin_types.go

```diff
@@ -41,3 +33,61 @@ type NoBoardFoundError struct {
 func (err NoBoardFoundError) Error() string {
 	return fmt.Sprintf("could not determine %q model", err.modelName)
 }
+
+// PinDefinition describes a gpio pin on a linux board.
+type PinDefinition struct {
```

Comment:
I'd prefer to move all the new code up to where the old code had been. I think you're right to keep `PinDefinition`, `PinDefinitions`, `UnmarshallJSON`, and `Validate` all in a contiguous block, but I'd like `PinDefinition` to be defined before it is used, and it's used up on line 24.